spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -79,7 +80,14 @@ public ConsumerDestination provisionConsumerDestination(String topicName, String
 
 		String subscriptionName;
 		Subscription subscription;
-		if (StringUtils.hasText(group)) {
+		if (StringUtils.hasText(properties.getExtension().getCustomSubscription())) {
+			String customName = properties.getExtension().getCustomSubscription();
+			subscriptionName = ProjectSubscriptionName.isParsableFrom(customName) ?
```

Comment:
I think allowing fully qualified subscription names here could be problematic when combined with auto-creation of resources because the auto-created subscription would be in the wrong project.
However, I think you will avoid the problem if you don't parse out the subscription short name, but keep it as fully qualified.

Reply:
> when combined with auto-creation of resources

This is the the code that does the auto-provisioning. The FR was specifically to not do the auto-provisioning and use a pre-existing subscription (I expanded the requirement to be potentially outside the default GCP project)

Reply:
Removed.